hyperemitter.js

```diff
@@ -106,21 +117,17 @@ function HyperEmitter (db, schema, opts) {
 }
 
 HyperEmitter.prototype.emit = function (name, data, cb) {
-  var encoder = headers[name] || this.messages[name]
-  var err
+  var encoder = this.codecs[name]
 
-  if (!encoder) {
-    err = new Error('Non supported event')
-    if (cb) { return cb(err) }
-    else throw err
-  }
+  if (encoder) data = encoder.encode(data)
+  else this.status.emit('no encoder found for ' + name)
```

Comment:
don't emit something like this.

I think that:

1. if the argument is a String or a Buffer, then we can send them through as Buffers, and emit them as such.
2. if it's an object, then we should return an error, just as before.